cpp/src/arrow/csv/converter.cc

```diff
@@ -183,6 +187,57 @@ Status FixedSizeBinaryConverter::Convert(const BlockParser& parser, int32_t col_
   return Status::OK();
 }
 
+/////////////////////////////////////////////////////////////////////////
+// Concrete Converter for booleans
+
+class BooleanConverter : public ConcreteConverter {
+ public:
+  using ConcreteConverter::ConcreteConverter;
+
+  Status Convert(const BlockParser& parser, int32_t col_index,
+                 std::shared_ptr<Array>* out) override;
+
+ protected:
+  Status Initialize() override {
+    // TODO no need to build separate Tries for each BooleanConverter instance
+    RETURN_NOT_OK(InitializeTrie(options_.true_values, &true_trie_));
+    RETURN_NOT_OK(InitializeTrie(options_.false_values, &false_trie_));
+    return ConcreteConverter::Initialize();
+  }
+
+  Trie true_trie_;
+  Trie false_trie_;
+};
+
+Status BooleanConverter::Convert(const BlockParser& parser, int32_t col_index,
+                                 std::shared_ptr<Array>* out) {
+  BooleanBuilder builder(type_, pool_);
+
+  auto visit = [&](const uint8_t* data, uint32_t size, bool quoted) -> Status {
+    // XXX should quoted values be allowed at all?
```

Comment:
Generators could unconditionally emit quoted CSV and still be legal; I think quoting should be transparent to conversion

Reply:
Right. It's an open question... If someone complains, we may add a setting :-)